full-service/src/service/transaction.rs

```diff
@@ -351,7 +383,26 @@ where
                 builder.select_txos(&conn, max_spendable)?;
             }
 
-            let tx_proposal: TxProposal = builder.build(&conn)?;
+            let tx_proposal = match memo {
+                TransactionMemo::RTH => {
+                    let mut memo_builder = RTHMemoBuilder::default();
+                    memo_builder
+                        .set_sender_credential(SenderMemoCredential::from(&from_account_key));
+                    memo_builder.enable_destination_memo();
+                    builder.build(Some(Box::new(memo_builder)), &conn)?
```

Comment:
This can be moved out of the match statement, the match statement can construct the boxed memo builder.